src/app/route.ts

```diff
@@ -31,64 +34,97 @@ const missingParameterResponse = (parameterName: string) =>
+  const fiatAmount = await readFromBodyOrQueryParams(req, body, "fiatAmount") || "50"
+  const fiatCurrency = await readFromBodyOrQueryParams(req, body, "fiatCurrency") || "EUR"
+  const defaultFiatAmount = await readFromBodyOrQueryParams(req, body, "defaultFiatAmount") || "0"
+  const defaultCryptoAmount = await readFromBodyOrQueryParams(req, body, "defaultCryptoAmount") || "0"
+  const cryptoCurrencyCode = await readFromBodyOrQueryParams(req, body, "cryptoCurrencyCode") || "NEAR"
+  const environment = await readFromBodyOrQueryParams(req, body, "environment") || "staging"
 
   if (!apiKey) {
     return missingParameterResponse("apiKey")
   }
-
+  if (!apiSecret) {
+    return missingParameterResponse("apiSecret")
+  }
   if (!walletAddress) {
     return missingParameterResponse("walletAddress")
   }
-
   if (!siloEngineAccountId) {
     return missingParameterResponse("silo")
   }
 
   const targetAddress = getAddress(walletAddress)
+  const forwarderAddress = await getForwarderAddress(targetAddress, siloEngineAccountId)
 
-  const url = new URL("https://global.transak.com/")
-
-  url.searchParams.set("network", "near")
-  url.searchParams.set("apiKey", apiKey)
-  url.searchParams.set("disableWalletAddressForm", "true")
-  url.searchParams.set(
-    "walletAddress",
-    await getForwarderAddress(targetAddress, siloEngineAccountId),
-  )
+  const savedAccessToken = await getAccessToken(apiKey)
 
-  if (fiatAmount) {
-    url.searchParams.set("fiatAmount", fiatAmount)
+  const accessToken = 
+    savedAccessToken ?? 
+    await refreshTransakToken(apiSecret, environment, apiKey)
+      
+  if (!accessToken) {
+    return new NextResponse("Failed to obtain access token", { status: 401 })
   }
 
-  if (fiatCurrency) {
-    url.searchParams.set("fiatCurrency", fiatCurrency)
+  const widgetRequestData = {
+    "widgetParams": {
+      "apiKey": apiKey,
+      "referrerDomain": "auroracloud.dev",
+      "fiatAmount": fiatAmount,
+      "defaultFiatAmount": defaultFiatAmount,
+      "defaultCryptoAmount": defaultCryptoAmount,
+      "fiatCurrency": fiatCurrency,
+      "cryptoCurrencyCode": cryptoCurrencyCode,
+      "walletAddress": forwarderAddress,
+    },
   }
+ 
+  const widgetUrl = await createTransakSession(environment, widgetRequestData, accessToken)
 
-  if (defaultFiatAmount) {
-    url.searchParams.set("defaultFiatAmount", defaultFiatAmount)
+  if (!widgetUrl) {
+    return new NextResponse("Failed to create Transak session", { status: 500 })
   }
 
-  if (defaultCryptoAmount) {
-    url.searchParams.set("defaultCryptoAmount", defaultCryptoAmount)
+  // If it is a POST request, return JSON
+  if (req.method === "POST") {
+    return NextResponse.json({ widgetUrl })
   }
 
-  if (environment) {
-    url.searchParams.set("environment", environment)
-  }
+  // If it is a GET request, redirect
+  const url = new URL(widgetUrl)
 
   return NextResponse.redirect(url.href)
 }
+
+export { handleRequest as GET, handleRequest as POST }
```

Comment:
We could add new lines to the bottom of all of these files.

Reply:
I realized I had prettier disabled since that first PR. Re-enabled it and updated all files with proper formatting